Improve performance of jaeger to internal traces translation

dmitryax opened the pull request:
Results of BenchmarkProtoBatchToInternalTraces:

Before:
```
BenchmarkProtoBatchToInternalTraces-12        329313          3346 ns/op        3352 B/op         43 allocs/op
```

After:
```
BenchmarkProtoBatchToInternalTraces-12        646017          1737 ns/op        2064 B/op         32 allocs/op
```

Roughly 80% of improvement was gained just by avoiding using intermediate map structure before creation of AttributeMap.

bogdandrutu commented on consumer/pdata/common.go:
```diff
@@ -260,6 +260,14 @@ func (am AttributeMap) InitFromMap(attrMap map[string]AttributeValue) AttributeM
 }
 
+// InitEmptyWithCapacity constructs an empty AttributeMap with predefined slice capacity.
+func (am AttributeMap) InitEmptyWithCapacity(cap int) {
```
Can we add this in a separate PR? There are more places where we need to update and use this.

tigrannajaryan commented:
> Roughly 80% of improvement was gained just by avoiding using intermediate map structure before creation of AttributeMap.

Nice!

Can you also post E2E before/after test results so that we can see what's the overall impact?

dmitryax commented:
e2e tests results:

Before:
```
Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
 
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
 
Trace10kSPS/JaegerGRPC                  |PASS  |     15s|    30.7|    32.3|         44|         54|    149900|        149900|
```

After:
```
Test                                    |Result|Duration|CPU Avg%|CPU Max%|RAM Avg MiB|RAM Max MiB|Sent Items|Received Items|
 
----------------------------------------|------|-------:|-------:|-------:|----------:|----------:|---------:|-------------:|
 
Trace10kSPS/JaegerGRPC                  |PASS  |     15s|    28.2|    29.3|         46|         56|    149900|        149900|
```

dmitryax commented:
@bogdandrutu I removed the capacity pre-initialization from this PR. I'll create another one once this is merged. PTAL

Updated benchmark results _after_:
```
BenchmarkProtoBatchToInternalTraces-12    	  497078	      2312 ns/op	    2096 B/op	      35 allocs/op
```
